readme.md

```diff
@@ -4,38 +4,69 @@ A rust version SPDM implementation.
 
 It is derived from https://github.com/DMTF/libspdm.
 
+## Documentation
+All documents are put at [doc](./doc/) folder.
+
 ## Build Rust SPDM
 
+### Checkout repo
+```
+git clone https://github.com/jyao1/rust-spdm.git
+git submodule update --init --recursive
+```
+
 ### Tools
 
 1. Install [RUST](https://www.rust-lang.org/)
 
-please use nightly-2022-04-07.
+Please use nightly-2022-04-07.
 
 2. Install [NASM](https://www.nasm.us/)
 
 Please make sure nasm can be found in PATH.
 
-3. Install LLVM
+3. Install [LLVM](https://llvm.org/)
 
 Please make sure clang can be found in PATH.
 
+4. Install [Perl](https://www.perl.org/)
+
+    1.	This is for crate ring
+    2.	This is for windows, linux should have shipped perl by default

```

Comment:
why you use "should have done" ?

Do you mean it does ship or does not ship?

Reply:
> why you use "should have done" ?
> 
> Do you mean it does ship or does not ship?

Because Linux has many distributions, I am not sure if there is a distribution that has no perl included by default.

Reply:
The problem is "subjunctive mood".

What you actually mean is "Linux does not ship perl by default" -- is this your purpose?

Reply:
> The problem is "subjunctive mood".
> 
> What you actually mean is "Linux does not ship perl by default" -- is this your purpose?

No, it is not my purpose,
I got what you mean, sorry for this kind of error, will fix soon.